### DISCUSSION TITLE
Botched upgrade from 2.4 to latest v2 (2.6.8)

### DISCUSSION THREAD

**Discussion by petersonespacoporto** [wallabag/wallabag#7219] [Q&A]
So I noticed the docs had instructions to upgrade from 2.5 to 2.6, but not from 2.4 to 2.5... I don't know if I had to do anything extra, but here's what I did (shared hosting):

- backed up the database;
- copied files to wallabag's directory (retaining my parameters.yml);
- I actually changed the parameters.yml to implement the mailer_dsn thing;
- tried to run "bin/console cache:clear --env=prod". It warned me that I was running an older-than-supported version of PHP. I upgraded my PHP, then I ran it again. This time it threw out an error:

`Fatal error: Declaration of Wallabag\UserBundle\Controller\RegistrationController::registerAction(Symfony\Component\HttpFoundation\Request $request) must be compatible with FOS\UserBundle\Controller\RegistrationController::registerAction(Symfony\Component\HttpFoundation\Request $request): Symfony\Component\HttpFoundation\Response in /home/petercast/www/salvos/src/Wallabag/UserBundle/Controller/RegistrationController.php on line 10`

Now I already imagined something went wrong here but I tried to go through with it just in case - it's not like I knew how to fix this anyway...

- So I tried to run the SQL queries - [the ones recommended when upgrading 2.5 to 2.6](https://doc.wallabag.org/en/admin/query-upgrade-25-26). The first and the third went OK (I mean, at least they didn't throw any errors). The second one, however, gave me this:

`#1064 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'JSON DEFAULT NULL' at line 1 `

Now accessing my wallabag instance gives me a blank page - which, of course, I expected, given these errors.

Could anyone help me fix my install? :) Thanks in advance!

**Comment by petersonespacoporto** [comment]
Ok So I made _some_ progress

I compared the files at src/Wallabag/UserBundle/Controller in my server with the release's tarball, and found that there were two extra (therefore legacy) files in my server. I backed them up locally and removed them from the server. Now the previous error is gone but I've got a new one, which is rather visually different:

```
In DefinitionErrorExceptionPass.php line 54:
                                                                               
  Class "Doctrine\DBAL\Driver\PDOPgSql\Driver" not found while loading "Wallabag\CoreBundle\Doctrine\DBAL\Driver\CustomPostgreSQLDriver".    
```             
                                                                               
(The second line is in fact inside a red rectangle)

Plus accessing my install no longer gives me a blank screen; it's a 503 Service Unavailable message. Yay?

Any help would be much appreciated!

**Comment by petersonespacoporto** [comment]
Ok, I don't know why I hadn't tried this first, but it turns out again, even though the error was different, this was a matter of extraneous files!!

I removed the "DBAL" folder from src/Wallabag/CoreBundle/Doctrine and tried again; I got another error referring to "Wallabag\CoreBundle\Helper\DetectActiveTheme" and removed that too.

Then I tried again and...

```
// Clearing the cache for the prod environment with debug                      
// false                                                                       
                                                                                
[OK] Cache for the "prod" environment (debug=false) was successfully cleared.
```

Which seems alright!!

Since I think the 503 error is due to maintenance in my shared host server, I'll try again later and see if it's alright. Still concerned about the MySQL query error though, if it doesn't work it might be due to that

**Comment by petersonespacoporto** [comment]
Ok so I basically reverted to version 4.1 for the time being. If anyone can point out what I did wrong I might one day try upgrading again :)

**Comment by j0k3r** [comment]
> to upgrade from 2.5 to 2.6, but not from 2.4 to 2.5...
Because there is no particular stuff to do.
Upgrading is usually just `make update`

> Now I already imagined something went wrong here but I tried to go through with it just in case - it's not like I knew how to fix this anyway...

Have you tried to clear the cache? `rm -rf /var/cache`

**Reply by petersonespacoporto** [reply]
Yes, I didn't say it in the original post but I did do it

Like, is this even allowed? Sorry if one should obviously not try to skip versions, I thought it was ok

In any case I needed PHP 8+ so I managed to upgrade to 2.4.3 at least! Haven't tried 2.5 though
